### PR TITLE
(REVIEW ONLY) ENG-7968: Initial wiring together of DRBufferSizeTuner

### PR DESCRIPTION
**Pull request by afrogers**

### CODE REVIEW

File: src/ee/storage/DRTupleStream.cpp [hunk 1]
```diff
@@ -342,12 +354,19 @@ void DRTupleStream::endTransaction() {
      m_uso += io.position();
 
      m_opened = false;
+
+    uint32_t bufferRowCount = m_currBlock->updateRowCountForDR(m_txnRowCount);
+    if (m_rowTarget >= 0 && bufferRowCount >= m_rowTarget) {
```
Comment (wweiss-voltdb) [inline src/ee/storage/DRTupleStream.cpp]: Just out of curiosity, if m_rowTarget == 0 and bufferRowCount == 0, what will happen?

File: src/ee/common/Topend.h [hunk 2]
```diff
@@ -62,7 +62,7 @@ class Topend {
             bool sync,
             bool endOfStream) = 0;
 
-    virtual void pushDRBuffer(int32_t partitionId, StreamBlock *block) = 0;
+    virtual int64_t pushDRBuffer(int32_t partitionId, StreamBlock *block) = 0;
```
Comment (wweiss-voltdb) [inline src/ee/common/Topend.h]: I don't see the update for voltdbipc.cpp. I assume it is in the next commit?

Comment (afrogers) [inline src/ee/common/Topend.h]: Nice catch. *shakes fist at voltdbipc*

**Comment by nshi** [comment]
Need to update README.thirdparty to include the additional library being added here.

File: src/ee/storage/DRTupleStream.cpp [hunk 3]
```diff
@@ -342,12 +354,19 @@ void DRTupleStream::endTransaction() {
...
+        extendBufferChain(0);
+    }
+    m_txnRowCount = 0;
 }
 
 // If partial transaction is going to span multiple buffer, first time move it to
 // the next buffer, the next time move it to a 45 megabytes buffer, then after throw
 // an exception and rollback.
 bool DRTupleStream::checkOpenTransaction(StreamBlock* sb, size_t minLength, size_t& blockSize, size_t& uso) {
+    // Hack:
```
Comment (nshi) [inline src/ee/storage/DRTupleStream.cpp]: Hack!

**Pull request closed without merging**